vividus-extension-selenium/src/main/java/org/vividus/ui/util/LocatorConversionUtils.java

```diff
@@ -86,31 +91,46 @@ public Locator convertToLocator(String locatorAsString)
 
     private void applyFilter(Locator locator, String filterType, String filterValue)
     {
-        createLocator(elementActionService.getFilterLocatorTypes(),
-            type -> locator.addFilter(type, filterValue), "filter", filterType);
+        createLocator(elementActionService.getFilterLocatorTypes(), locator::addFilter, "filter", filterType,
+                filterValue);
     }
 
     private Locator convertToLocator(String searchType, String searchValue)
     {
-        return createLocator(elementActionService.getSearchLocatorTypes(),
-            type -> new Locator(type, searchValue), "locator", searchType);
+        return createLocator(elementActionService.getSearchLocatorTypes(), Locator::new, "locator", searchType,
+            searchValue);
     }
 
-    private static Locator createLocator(Set<LocatorType> locatorTypes, Function<LocatorType, Locator> mapper,
-            String operatorType, String searchType)
+    private Locator createLocator(Set<LocatorType> locatorTypes, BiFunction<LocatorType, String, Locator> mapper,
+            String operatorType, String searchType, String searchValue)
     {
         return findLocatorType(locatorTypes, searchType)
-                .map(mapper)
+                .map(t -> mapper.apply(t, searchValue))
+                .or(() -> dynamicLocators.getNullable(searchType)
+                                         .map(l -> createLocator(locatorTypes, mapper, operatorType, l.getLocatorType(),
+                                                 buildLocator(searchValue, l))))
                 .orElseThrow(() -> new IllegalArgumentException(
-                    String.format("Unsupported %s type: %s", operatorType, searchType)));
+                        String.format("Unsupported %s type: %s", operatorType, searchType)));
+    }
+
+    private String buildLocator(String searchValue, LocatorPattern locatorPattern)
+    {
+        String[] params =
+            (searchValue.contains("\\,") ? ESCAPED_PARAMS_SEPARATOR : PARAMS_SEPARATOR).split(searchValue);
+        return String.format(locatorPattern.getPattern(), params);
     }
 
     private static Optional<LocatorType> findLocatorType(Set<LocatorType> locatorTypes, String type)
     {
         String typeInLowerCase = type.toLowerCase();
         return locatorTypes.stream()
-                    .filter(t -> StringUtils.replace(t.getKey().toLowerCase(), "_", "")
-                            .equals(typeInLowerCase))
-                    .findFirst();
+                           .filter(t -> StringUtils.replace(t.getKey().toLowerCase(), "_", "")
+                           .equals(typeInLowerCase))
+                           .findFirst();
```

Comment:
noisy changes

Reply:
It is bad formatting :) 
What about: “Always leave the campground cleaner than you found it” ???